blank_rust_project/src/main.js

```diff
@@ -62,14 +62,29 @@ function signedInFlow() {
     // Displaying the signed in flow container.
     document.getElementById('signed-in-flow').classList.remove('d-none');
 
-    window.contract.welcome({account_id:window.accountId}).then(response => document.getElementById('speech').innerText = response.text);
+    getGreeting();
 
     // Adding an event to a sign-out button.
     document.getElementById('sign-out-button').addEventListener('click', () => {
         walletAccount.signOut();
         // Forcing redirect.
         window.location.replace(window.location.origin + window.location.pathname);
     });
+
+    // Adding an event to change greeting button.
+    document.getElementById('change-greeting').addEventListener('click', () => {
+        setGreeting();
+    });
+}
+
+async function setGreeting() {
```

Comment:
Is there a reason why message is not passed as an argument?

Reply:
A good argument for that is simplifying code. It's usually safe to assume that devs know how to use basic javascript, so they don't need an example of how to pass user input from the webpage. This allows less overall code and more usefulness per line of example code (which is definitely a metric we want to optimize for)